pkg/envtest/server.go

```diff
@@ -32,6 +35,8 @@ const (
 	envEtcdBin             = "TEST_ASSET_ETCD"
 	envKubectlBin          = "TEST_ASSET_KUBECTL"
 	envKubebuilderPath     = "KUBEBUILDER_ASSETS"
+	envStartTimeout        = "KUBEBUILDER_CONTROLPLANE_START_TIMEOUT"
+	envStopTimeout         = "KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT"
```

Comment:
I would recommend making these parts of `Environment` struct below as public args, so that callers can override the defaults and lets tweak the defaults on the basis of your experience. Lets avoid passing these through environment variables.

Reply:
hey @droot , thanks for your feedback.

While I think adding them to the struct is a good idea I believe it still makes sense to populate them from the environment, because a controller-manager may contain more than one controller and this is a setting one does not want to set per controller but globally for all controllers, since there is no reason to believe that the test apiserver and etcd will have a different startup time per controller.

What do you think about adding them to the environment struct but populating them from environment variables in case they are unset?

Reply:
>What do you think about adding them to the environment struct but populating them from environment variables in case they are unset?

Picking the defaults (if unspecified) from environment variables sounds reasonable to me.

Reply:
@droot I've updated the PR to add the start/stop timeout options to the `Environment` struct and only default from environment variables if they are unspecified, please take another look